File: BookSpaceApp/src/main/java/com/example/bookspaceapp/model/Review.java
package com.example.bookspaceapp.model;
import javax.persistence.*;
import javax.validation.constraints.*;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;

@Getter
@Setter
@Entity
@ToString
@AllArgsConstructor
@NoArgsConstructor

@Table(name = "review")
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "review")
    @Size(max = 600, message = "Book review can have maximum 600 characters!")
    private String review;

    @Column(name = "date")
//    @JsonFormat(pattern = "yyyy-MM-dd")
//    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private LocalDate date;

    /** MANY TO ONE relation with Book **/
    @ManyToOne
    @JoinColumn(name = "book_id")
    @JsonIgnore
    private Book book;

    /** MANY TO ONE relation with User **/
    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User user;

    public Review(String review) {
        this.review = review;
    }
}

File: bookspace/src/app/model/review.ts
import { User } from "./user";


 export class Review {
    public id: number;
   public review: string;
   public date: string;
   public user: User;
  //  public userId: number;
  //  public bookId: number;

  
 
   constructor() {
      this.id = null as any;
     this.review = '';
     this.date = '';
    //  this.userId = null as any;
    //  this.bookId = null as any;
    this.user = new User();

     
   }
 
 }